### PR TITLE
Feature/sn trigger

### PR DESCRIPTION
**Pull request by ast0815**
Another PR, just to enable discussion of the code.

Based on #30, so it includes all of those commits as well.

### CODE REVIEW

File: UserTools/dimfit/dimfit.cpp [hunk 1]
```diff
@@ -26,16 +26,18 @@ bool dimfit::Initialise(std::string configfile, DataModel &data){
...
+    Log("WARN: No m_time_window parameter found. Using a value of 20 (seconds)", WARN, verbose);
   }
-  time_window_ns = time_window_s * 1E9;
-  if(!m_variables.Get("time_window_step", time_window_step_s)) {
+  m_time_window = time_window_s * TimeDelta::s;
+  double time_window_step_s;
+  if(!m_variables.Get("m_time_window_step", time_window_step_s)) {
     time_window_step_s = 0.2;
-    Log("WARN: No time_window_step parameter found. Using a value of 0.2 (seconds)", WARN, verbose);
+    Log("WARN: No m_time_window_step parameter found. Using a value of 0.2 (seconds)", WARN, verbose);
```
Comment (tdealtry) [inline UserTools/dimfit/dimfit.cpp]: Config file parameters shouldn't have `m_` in front of them. Please remove from the `Get` and `Log` functions

**Comment by tdealtry** [comment]
https://github.com/ast0815/TriggerApplication/compare/feature/sub_sample_times...ast0815:feature/SN-trigger for others interested in reviewing
Anyway, looks good apart from the minor comment

File: UserTools/dimfit/dimfit.cpp (hunk 1 above)
Comment (ast0815) [reply UserTools/dimfit/dimfit.cpp]: Right. The search and replace was a bit greedy there. ^^

**Comment by brichards64** [comment]
Ok again discussion should be had in issues and PRs only made when your ready to merge them.

There are also a number of conflicts here. Is this a real PR or can i close it?

**Comment by ast0815** [comment]
This is not ready yet. Hence the "WIP" in the title.

Why not discuss code in a PR during development? Having this as a PR instead of an issue makes it easier to actually comment on the code changes.

**Comment by brichards64** [comment]
I still prefer issues rather than PRs but iv made a new label for work in progress so i can ignore these. please remove the label when ready

**Comment by tdealtry** [comment]
Actually I quite like the WIP, because of being able to see & comment on ideas of code changes inline (I know it's not good for package manager "inbox 0" OCD)

**Comment by brichards64** [comment]
Yeh i know what you mean Tom.

I think i can make my peace with it now iv implemented the neon turquoise label so when im scanning down the open PRs i can see it easily.

**Comment by ast0815** [comment]
Split of two topical PRs ready for merging: #52 #53

**Comment by ast0815** [comment]
This is ready now. PRs 68 through 71 split some changes into smaller chunks for easier review. This last bit still depends on WCSim/WCSIm#280.

File: UserTools/SupernovaDirectionCalculator/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# SupernovaDirectionCalculator
```
Comment (tdealtry) [inline UserTools/SupernovaDirectionCalculator/README.md:1]: Can you fill in the README please? Brief description of what it does, what it reads from the data model (related to what tools need to run first), and what the config options are

File: UserTools/SupernovaDirectionCalculator/SupernovaDirectionCalculator.cpp [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+  return true;
+}
+
+
+bool SupernovaDirectionCalculator::Execute(){
+
+  if(m_stopwatch) m_stopwatch->Start();
+
+  //// YOUR CODE HERE
```
Comment (tdealtry) [inline UserTools/SupernovaDirectionCalculator/SupernovaDirectionCalculator.cpp:73]:
```suggestion
```

File: UserTools/SupernovaDirectionCalculator/SupernovaDirectionCalculator.cpp [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+
+bool SupernovaDirectionCalculator::Finalise(){
+
+  if(m_stopwatch) {
+    Log(m_stopwatch->Result("Execute", m_stopwatch_file), INFO, m_verbose);
+    m_stopwatch->Start();
+  }
+
+  //// YOUR CODE HERE
```
Comment (tdealtry) [inline UserTools/SupernovaDirectionCalculator/SupernovaDirectionCalculator.cpp:122]:
```suggestion
```

File: configfiles/WCSimBONSAI/BONSAIToolConfig [hunk 5]
```diff
@@ -2,4 +2,5 @@ use_stopwatch 1
 #stopwatch_file bonsai_stopwatch.pdf
 verbose 3
 nhitsmin 1
-nhitsmax 2000
+nhitsmax 1000
+use_stopwatch 1
```
Comment (tdealtry) [inline configfiles/WCSimBONSAI/BONSAIToolConfig:6]:
```suggestion
```
Remeber to pull from #71

**Comment by ast0815** [comment]
I have split off PR #72. It should not need the double-only WCSim version.

**Comment by tdealtry** [comment]
@brichards64 can you trigger a build? This _should_ pass it now that WCSim has been updated (and it doesn't depend on the docker base image - Travis does a pull on WCSim). It seems that this is actually __required__ now (thanks for testing @tadoyle!)

**Pull request merged**